editor/components/block-settings-menu/block-duplicate-button.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import { flow, noop, last, every } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { IconButton, withContext } from '@wordpress/components';
+import { compose } from '@wordpress/element';
+import { cloneBlock, getBlockType } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import { getBlocksByUid, getBlockIndex } from '../../store/selectors';
+import { insertBlocks } from '../../store/actions';
+
+export function BlockDuplicateButton( { blockNames, onDuplicate, onClick = noop, isLocked, small = false } ) {
+	const canDuplicate = every( blockNames.map( name => {
+		const type = getBlockType( name );
+		return ! type.useOnce;
+	} ) );
+	if ( isLocked || ! canDuplicate ) {
```

Comment:
Minor (since it's not a common scenario): if `isLocked` is `true` here, we've wasted effort calculating `canDuplicate`, when we could have done an early return before the `every`.